samples/rust/blinky_rust/src/lib.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2024 Linaro LTD
+// SPDX-License-Identifier: Apache-2.0
+
+#![no_std]
+
+use log::warn;
+
+use core::ffi::c_void;
+
+use zephyr::sys::gpio::GPIO_OUTPUT_ACTIVE;
+
+#[no_mangle]
+extern "C" fn rust_main() {
+    zephyr::set_logger();
+
+    warn!("Starting blinky");
+    // println!("Blinky!");
+    // Invoke "blink" as a user thread.
+    // blink();
+    if false {
+        unsafe {
+            zephyr::raw::k_thread_user_mode_enter
+                (Some(blink),
+                core::ptr::null_mut(),
+                core::ptr::null_mut(),
+                core::ptr::null_mut());
+        }
+    } else {
+        unsafe {
+            blink(core::ptr::null_mut(),
+                  core::ptr::null_mut(),
+                  core::ptr::null_mut());
+        }
+    }
+}
+
+// fn blink() {
+unsafe extern "C" fn blink(_p1: *mut c_void, _p2: *mut c_void, _p3: *mut c_void) {
+    warn!("Inside of blinky");
+
+    let mut led0 = zephyr::devicetree::aliases::led0::get_instance();
```

Comment:
Can we somehow make use of `Option` types, similar to how `DEVICE_DT_GET_OR_NULL` would work? This would allow to re-use code for different platforms where runtime checks are forced if a device tree node would be available or not.

Reply:
That seems reasonable, but I'll have to think about how that would work. I do like the idea of the DT being actually present as modules, but I'm not aware of a way to query that, and it doesn't really make sense to query the existence of modules at compile time.